### ISSUE TITLE
Cerebral (or function-tree) catch my errors

### ISSUE THREAD

**Issue by etienne-dldc** [cerebral/cerebral#520]
Hi, :)

## The problem ⚠️ 

When I throw an error inside an async action (that return a Promise), the error is caught by Cerebral or Function tree and the error just disappear but the signal that contains the action does not continue and blink forever in the debugger.

## Possible solution 🤔 

I think putting some `setTimeout()` with no time (wait for the callback to be empty) might solve this issue, but I'm not sure where this `setTimeout` should go....

## Infos ℹ️ 

I'm using the last version of cerebral : `cerebral@next` which resolve to `2.0.0-0-alpha.fe8128f5`

**Comment by christianalfoni** [comment]
@etienne-dldc Hm, again, thanks for a great description :)

Hm, I thought we fixed this:

*Controller.js*
```js
    this.runTree(name, signal, payload || {}, (error) => {
      if (error) {
        throw error
      }
    })
```

It should be safe to run function tree and get all errors. So if not the error is caught and rethrown here, something is wrong. Could you verify that the callback there in Controller does not get the error? Ah... wait... of course, it is probably swallowed again by the promise.

Okay, cause there is async error throwing in function tree already: `runTree.on('error'....`, I did that first, but changed it, but forgot about this. I can change it back :)

**Comment by etienne-dldc** [comment]
When I activate the "Pause On Caught Exeption" in Chrome I see the error thrown by the code you quote above but since a promise is executed somewhere in the call stack, the Promise caught the error and eat it (I guess)...

**Comment by Guria** [comment]
Is it fixed with #525 ?

**Comment by christianalfoni** [comment]
Jup :)
